<!-- ROSI URDF MODEL -->
<robot name="rosi">

	<!-- Defining arm links -->
	<link name="base_link">
		<origin xyz="0 0 0" rpy="0 0 0"/> 
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/rosi_base.stl" />
			</geometry>
		</visual>
	</link>

	<link name="arm_1">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/arm.stl" />
			</geometry>
			<origin xyz="0,1805 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<link name="arm_2">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/arm.stl" />
			</geometry>
			<origin xyz="-0,1805 0 0" rpy="0 0 3,14" />
		</visual>
	</link>

	<link name="arm_3">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/arm.stl" />
			</geometry>
			<origin xyz="0,1805 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<link name="arm_4">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/arm.stl" />
			</geometry>
			<origin xyz="-0,1805 0 0" rpy="0 0 3,14" />
		</visual>
	</link>

	<!-- Defining wheel links -->
	<link name="wheel_1">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/wheel.stl" />
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<link name="wheel_2">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/wheel.stl" />
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<link name="wheel_3">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/wheel.stl" />
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<link name="wheel_4">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/wheel.stl" />
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0" />
		</visual>
	</link>

	<!-- Defining sensors links -->

	<link name="velodyne">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/velodyne.stl" />
			</geometry>
		</visual>
	</link>

	<link name="kinect">
		<visual>
			<geometry>
				<mesh filename="package://rosi_defy/urdf/meshes/kinect.stl" />
			</geometry>
			<origin xyz="-0.023 0 0" rpy="-1,57 0 1,57" />
		</visual>
	</link>

	<!-- link name="velodyne" /-->
	<!-- link name="kinect" /-->

	<!-- Defining arms joints -->
	<joint name="joint_arm_1" type="continuous">
		<parent link="base_link" />
		<child link="arm_1" />
		<origin xyz="0.3152 -0.2210 -0.0993" rpy="1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_arm_2" type="continuous">
		<parent link="base_link" />
		<child link="arm_2" />
		<origin xyz="-0.3152 -0.2210 -0.0993" rpy="1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_arm_3" type="continuous">
		<parent link="base_link" />
		<child link="arm_3" />
		<origin xyz="0.3152 0.2210 -0.0993" rpy="-1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_arm_4" type="continuous">
		<parent link="base_link" />
		<child link="arm_4" />
		<origin xyz="-0.3152 0.2210 -0.0993" rpy="-1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<!-- Defining wheel joints -->
	<joint name="joint_wheel_1" type="continuous">
		<parent link="base_link" />
		<child link="wheel_1" />
		<origin xyz="0.3152 -0.2880 -0.0993" rpy="1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_wheel_2" type="continuous">
		<parent link="base_link" />
		<child link="wheel_2" />
		<origin xyz="-0.3152 -0.2880 -0.0993" rpy="1.57 0 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_wheel_3" type="continuous">
		<parent link="base_link" />
		<child link="wheel_3" />
		<origin xyz="0.3152 0.2880 -0.0993" rpy="-1.57 3.14 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_wheel_4" type="continuous">
		<parent link="base_link" />
		<child link="wheel_4" />
		<origin xyz="-0.3152 0.2880 -0.0993" rpy="-1.57 3.14 0" />
		<axis xyz="0 0 1" />
	</joint>

	<joint name="joint_velodyne" type="fixed">
		<parent link="base_link" />
		<child link="velodyne" />
		<origin xyz="0.242 0.002 0.1655" rpy="0 0.26 0" />
	</joint>

	<joint name="joint_kinect" type="revolute">
		<parent link="base_link" />
		<child link="kinect" />
		<origin xyz="0.3 0 0.009" rpy="1.57 1.57 0" />
		<limit effort="30" velocity="0.79" lower="-0.79" upper="0.79" />
		<axis xyz="0 0 1" />
	</joint>

</robot>